### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by mrBliss**
Highlights:

* A bunch of Shelley types are now parameterised by the crypto instead of the
  era. This means the instantiation changes from `StandardShelley` to
  `StandardCrypto`. See https://github.com/input-output-hk/ouroboros-network/pull/2702

* Because of changes to the era parameterisation in the ledger, one should now
  use `ShelleyBasedEra` (from `Ouroboros.Consensus.Shelley.Eras`) as the
  constraint on `era` instead of `Era`. With the former you will be able to
  derive many more constraints.

* Consensus supports a bunch of new queries:
  + `GetGenesisConfig`: return the (compacted) genesis config used by the node.
  + `DebugEpochState`: was previously called `GetCurrentEpochState`.
  + `DebugNewEpochState`: this contains slightly more information than the
    former `DebugEpochState`. NOTE: in the second commit we start using this to
    query the ledger state.
  + `DebugChainDepState`: contains the epoch nonces, etc.
  + `GetCurrentEra`: use
    ```haskell
    eraIndexToInt :: EraIndex xs -> Int
    ```
    on its result to get the current era as index starting from 0.

  Note that some of these queries require a new network version. You don't have
  to do anything for this. You will only notice this when submitting one of the
  new queries to an older node that doesn't support it; you'll get a
  `ShelleyEncoderUnsupportedQuery` exception. Better to get such an exception
  when submitting the unsupported query than getting no response at all because
  the receiver fails to decode the request.

  See https://github.com/input-output-hk/ouroboros-network/pull/2694 for more details.

### CODE REVIEW

File: cardano-cli/src/Cardano/CLI/Shelley/Orphans.hs
Comment (mrBliss) [inline cardano-cli/src/Cardano/CLI/Shelley/Orphans.hs:110]: The choice between `newtype` and `anyclass` determines whether the constructor name is included in the output or not, right? If the format of the ledger dump is important, then somebody should review this (and the other new instances below) more carefully.

Note that `PoolDistr` currently doesn't have a `Generic` instance, so we can't use `anyclass` for it.

**Comment by mrBliss** [comment]
This PR propagates the recent changes in the ledger and follow-up changes in consensus. Note that the Allegra and Mary eras are still identical to the Shelley era.

File: cardano-api/src/Cardano/Api/Typed.hs
Comment (Jimbo4350) [reply cardano-api/src/Cardano/Api/Typed.hs:4644]: Yep my next PR begins this process

File: cardano-api/src/Cardano/Api/Typed.hs
Comment (mrBliss) [inline cardano-api/src/Cardano/Api/Typed.hs]: Oh, I didn't realise the change was going to be so simple (one fewer constructor wrapping it) :slightly_smiling_face:.

**Comment by mrBliss** [comment]
@intricate I would squash https://github.com/input-output-hk/cardano-node/pull/2019/commits/83eff409fcbcf951f567c4d068f8f8bb3d40976d and https://github.com/input-output-hk/cardano-node/pull/2019/commits/eab4a742971ebc860f7561292cbc8e5bca8e77dd (the two "fix" commits) into https://github.com/input-output-hk/cardano-node/pull/2019/commits/8b6134dfe46f2b69e2301b604c7e5f88a9af5dd9 before merging.

**Review by intricate — APPROVED** [review]

**Review by dcoutts — APPROVED** [review]

**Comment by dcoutts** [comment]
bors merge

**Comment by mrBliss** [comment]
bors merge

**Comment by mrBliss** [comment]
bors merge

**Comment by mrBliss** [comment]
bors merge

**Pull request merged**
